cmd/kube-apiserver/app/server.go

```diff
@@ -544,6 +545,16 @@ func BuildAuthenticator(s *options.ServerRunOptions, extclient clientgoclientset
 		versionedInformer.Core().V1().Secrets().Lister().Secrets(v1.NamespaceSystem),
 	)
 
+	if c.EgressSelector != nil {
+		networkContext := egressselector.Master.AsNetworkContext()
+		var egressDialer utilnet.DialFunc
+		egressDialer, err = c.EgressSelector.Lookup(networkContext)
```

Comment:
nit: not sure we need all the intermediates

`egressDialer, err := c.EgressSelector.Lookup(egressselector.Master.AsNetworkContext()`